core/trino-main/src/main/java/io/trino/execution/ExecutionFailureInfo.java

```diff
@@ -175,4 +178,21 @@ else if (matcher.group(4) != null) {
         }
         return new StackTraceElement("Unknown", stack, null, -1);
     }
+
+    public static ExecutionFailureInfo rewriteTransportFailure(FailureDetector failureDetector, ExecutionFailureInfo executionFailureInfo)
+    {
+        if (executionFailureInfo.getRemoteHost() == null || failureDetector.getState(executionFailureInfo.getRemoteHost()) != GONE) {
```

Comment:
It is currently used in two places, `FaultTolerantStageScheduler` and `PipelinedStageExecution` to rewrite task failures related to a node crash. I wanted to avoid having a copy in both places. Also I didn't want to create a separate utility class for it. However since the logic is pretty straightforward it doesn't seem to be a big problem to have 2 copy of it. What do you think?